Kernel/Locking/Mutex.h

```diff
@@ -26,6 +28,12 @@ class Mutex {
 
 public:
     using Mode = LockMode;
+    // FIXME: Allow some customization
+    //  Note: Making this a template messes with Thread.h and our implementation
+    //        A fixme further down suggests passing this in a rank constructor,
+    //        That would then likely be the rank for the persistent rank change,
+    //        as spinlocks rely on their ranks being template arguments for now
```

Comment:
The spinlock templatization is intentional, maybe we can do something with CRTP for the mutex problem... I'm pretty sure we don't need dynamic ranks for Mutexes either.